### PR TITLE
Fix: Non-standard ports causes problem with links

### PR DESCRIPTION
**Pull request by skofman1**
Fixes: https://github.com/NuGet/NuGetGallery/issues/5217

### CODE REVIEW

File: src/NuGetGallery/UrlExtensions.cs [hunk 1]
```diff
@@ -1049,12 +1049,24 @@ internal static string EnsureTrailingSlash(string url)
 
             if (relativeUrl)
             {
-                return actionLink.Replace($"{protocol}://{hostName}", string.Empty);
+                return GetRelativeUrl(actionLink, protocol, hostName, url.RequestContext.HttpContext.Request.Url.Port, url.RequestContext.HttpContext.Request.Url.IsDefaultPort);
```
Comment (chenriksson) [inline src/NuGetGallery/UrlExtensions.cs]: nit: long line

File: src/NuGetGallery/UrlExtensions.cs [hunk 2]
```diff
@@ -1068,7 +1080,7 @@ internal static string EnsureTrailingSlash(string url)
 
             if (relativeUrl)
             {
-                return routeLink.Replace($"{protocol}://{hostName}", string.Empty);
+                return GetRelativeUrl(routeLink, protocol, hostName, url.RequestContext.HttpContext.Request.Url.Port, url.RequestContext.HttpContext.Request.Url.IsDefaultPort);
```
Comment (chenriksson) [inline src/NuGetGallery/UrlExtensions.cs]: nit: long line

File: src/NuGetGallery/UrlExtensions.cs [hunk 3]
```diff
@@ -1049,12 +1049,24 @@ internal static string EnsureTrailingSlash(string url)
...
+        {
+            if (!isDefaultPort)
+            {
+                return link.Replace($"{protocol}://{hostName}:{port}", string.Empty);
+            }
+            else
+            {
+                return link.Replace($"{protocol}://{hostName}", string.Empty);
+            }
+        }
```
Comment (chenriksson) [inline src/NuGetGallery/UrlExtensions.cs]: `VirtualPathUtility.ToAppRelative` doesn't work?

File: tests/NuGetGallery.Facts/UrlExtensionsFacts.cs [hunk 4]
```diff
@@ -3,6 +3,8 @@
 
 using NuGetGallery.Framework;
```
Comment (chenriksson) [inline tests/NuGetGallery.Facts/UrlExtensionsFacts.cs]: nit: sort usings

File: src/NuGetGallery/UrlExtensions.cs (hunk 3 above)
Comment (skofman1) [reply src/NuGetGallery/UrlExtensions.cs]: Tried... it doesn't :(

**Pull request merged**